pkg/sql/sem/tree/type_name.go

```diff
@@ -180,21 +182,35 @@ func ResolveType(
 
 // FormatTypeReference formats a ResolvableTypeReference.
 func (ctx *FmtCtx) FormatTypeReference(ref ResolvableTypeReference) {
-	if ctx.HasFlags(fmtStaticallyFormatUserDefinedTypes) {
-		switch t := ref.(type) {
-		case *types.T:
-			if t.UserDefined() {
+	switch t := ref.(type) {
+	case *types.T:
+		if t.UserDefined() {
+			if ctx.HasFlags(FmtAnonymize) {
+				ctx.WriteByte('_')
+				return
+			}
```

Comment:
don't you need an else here?

Reply:
well the previous case short-circuits, but i can add an else